src/components/DirectoryItemSelector/directory-item-selector.tsx

```diff
@@ -142,88 +133,70 @@ const DirectoryItemSelector: FunctionComponent<DirectoryItemSelectorProps> = ({
-                                              (val: any) => {
-                                                  // Accept every directory
-                                                  if (
-                                                      val.type ===
-                                                      ElementType.DIRECTORY
-                                                  ) {
-                                                      return true;
-                                                  }
-                                                  // otherwise filter with the custom itemFilter func
-                                                  return itemFilter(val);
-                                              }
-                                          )
-                                        : childrenWithMetadata;
-                                    // update directory content
-                                    addToDirectory(nodeId, children);
-                                });
-                        } else {
+                    if (
+                        childrenMatchedTypes.length > 0 &&
+                        equipmentTypes &&
+                        equipmentTypes.length > 0
+                    ) {
+                        fetchElementsInfos(
+                            childrenMatchedTypes.map((e: any) => e.elementUuid),
+                            types,
+                            equipmentTypes
+                        ).then((childrenWithMetadata) => {
+                            const children = itemFilter
+                                ? childrenWithMetadata.filter((val: any) => {
+                                      // Accept every directory
+                                      if (val.type === ElementType.DIRECTORY) {
+                                          return true;
+                                      }
+                                      // otherwise filter with the custom itemFilter func
+                                      return itemFilter(val);
+                                  })
+                                : childrenWithMetadata;
                             // update directory content
-                            addToDirectory(nodeId, childrenMatchedTypes);
-                        }
-                    })
-                    .catch((error) => {
-                        console.warn(
-                            `Could not update subs (and content) of '${nodeId}' : ${error.message}`
-                        );
-                    });
+                            addToDirectory(nodeId, children);
+                        });
+                    } else {
+                        // update directory content
+                        addToDirectory(nodeId, childrenMatchedTypes);
+                    }
+                })
+                .catch((error) => {
+                    console.warn(
```

Comment:
only a console log?

Reply:
How so ? Since it's in a catch bock I'd reckon a warn is warranted